devtools/regenerate_readmes.py

```diff
@@ -0,0 +1,53 @@
+r"""
+This script takes our talktorials and renders the first markdown cell
+of the notebook to the corresponding README.md in the containing folder.
+
+Usage
+-----
+
+.. code-block::
+
+    % python devtools/regenerate_readmes.py --output README.md \
+        teachopencadd/talktorials/001_query_chembl/talktorial.ipynb
+
+"""
+
+
+def first_markdown_cells(path, stopif=lambda cell: "## Theory" in cell["source"]):
+    import nbformat
+
+    nb = nbformat.read(path, nbformat.NO_CONVERT)
+    for cell in nb.cells:
+        if stopif(cell):
+            break
+        if cell["cell_type"] == "markdown":
+            yield cell["source"]
+
+
+def parse_cli():
+    from argparse import ArgumentParser
+
+    p = ArgumentParser()
+    p.add_argument("notebook", type=str)
+    p.add_argument("--output", type=str, default=None)
+    return p.parse_args()
+
+
+def main():
+    from pathlib import Path
+
+    args = parse_cli()
+
+    nbcontent = "\n\n\n".join(list(first_markdown_cells(args.notebook)))
+    if not nbcontent.strip():  # empty results
+        nbcontent = "> This talktorial is still under development"
```

Comment:
After some thought, I still do not understand why this works. The former T11 talktorials should have content in `nbcontent`, don't they? But still the README gets the "> This talktorial is still under development" content.

Reply:
By chance. We have a stop condition: if the cell contains `## Theory` then stop (because we are past the introduction). However, since T11 still has the old format where the learning goals have that theory header, if doesn't get processed and returns None.

Reply:
Ahh, yes, yes. In the old notebooks, we have "#**## Theory**" when listing the learning goals - in the new notebooks, we call this header "### Contents in *Theory*", which does not match the "## Theory" query! Very clever :)